packages/contentstack-auth/test/unit/commands/tokens-add.test.ts

```diff
@@ -100,3 +104,171 @@ describe('Tokens Add Command', () => {
+    });
+    it('Should throw an error if branch flag is passed along with stack not having branches enabled', async () => {
+      await TokensAddCommand.run([
+        '-m',
+        '-a',
+        'newToken',
+        '-k',
+        process.env.BRANCH_DISABLED_API_KEY!,
+        '-t',
+        process.env.BRANCH_DISABLED_MGMT_TOKEN!,
+        '--branch',
+        'main',
+      ]);
+      assert.calledOnce(errorStub);
+    });
+  });
+
+  describe('-Delivery token', () => {
+    it('Should ask for api key', async () => {
+      await TokensAddCommand.run(['-d', '-a', 'newToken']);
+      assert.calledWith(inquireStub, { type: 'input', message: 'CLI_AUTH_TOKENS_ADD_ENTER_API_KEY', name: 'apiKey' });
+    });
+    it('Should ask for delivery token', async () => {
+      await TokensAddCommand.run(['-d', '-a', 'newToken', '-k', process.env.BRANCH_ENABLED_API_KEY!]);
+      assert.calledWith(inquireStub, { type: 'input', message: 'CLI_AUTH_TOKENS_ADD_ENTER_TOKEN', name: 'token' });
+    });
+    it('Should ask for environment', async () => {
+      await TokensAddCommand.run([
+        '-d',
+        '-a',
+        'newToken',
+        '-k',
+        process.env.BRANCH_ENABLED_API_KEY!,
+        '-t',
+        process.env.BRANCH_ENABLED_DELIVERY_TOKEN!,
+      ]);
+      assert.calledWith(inquireStub, {
+        type: 'input',
+        message: 'CLI_AUTH_TOKENS_ADD_ENTER_ENVIRONMENT',
+        name: 'env',
+      });
+    });
+    it('Should add a new token if all the values are set correctly for stack with branches enabled', async () => {
+      await TokensAddCommand.run([
+        '-d',
+        '-a',
+        'newToken',
+        '-k',
+        process.env.BRANCH_ENABLED_API_KEY!,
+        '-t',
+        process.env.BRANCH_ENABLED_DELIVERY_TOKEN!,
+        '-e',
+        process.env.BRANCH_ENABLED_ENVIRONMENT!,
+      ]);
+      assert.calledWith(successStub, 'CLI_AUTH_TOKENS_ADD_SUCCESS');
+    });
+    it('Should add a new token if all the values are set correctly for stack with branches disabled', async () => {
+      await TokensAddCommand.run([
+        '-d',
+        '-a',
+        'newToken',
+        '-k',
+        process.env.BRANCH_DISABLED_API_KEY!,
+        '-t',
+        process.env.BRANCH_DISABLED_DELIVERY_TOKEN!,
+        '-e',
+        process.env.BRANCH_DISABLED_ENVIRONMENT!,
+      ]);
+      assert.calledWith(successStub, 'CLI_AUTH_TOKENS_ADD_SUCCESS');
+    });
+    it('Should throw and error for stack with branches disabled', async () => {
+      let branch = 'my-branch';
+      await TokensAddCommand.run([
+        '-d',
+        '-a',
+        'newToken',
+        '-k',
+        process.env.BRANCH_DISABLED_API_KEY!,
+        '-t',
+        process.env.BRANCH_DISABLED_DELIVERY_TOKEN!,
+        '-e',
+        process.env.BRANCH_DISABLED_ENVIRONMENT!,
+        '--branch',
+        branch,
+      ]);
+      assert.calledOnce(errorStub);
+    });
+  });
+});
```

Comment:
As discussed,

- please remove the deprecated flags and use the new flags and do not use any short flags.
- Check all test case execution times and verify if any test case taking longer time than expected if so increase the timeout.